src/sly_unpacker.cpp

```diff
@@ -1,53 +1,75 @@
 #include "types.hpp"
 #include "fs.hpp"
-#include "stream_utils.hpp"
 #include "wac.hpp"
 
 #include <algorithm>
 #include <fstream>
 #include <iterator>
 #include <stdexcept>
+#include <filesystem>
 
 int main(int argc, char *argv[]) {
     try {
-        if (argc <= 2)
+        if (argc < 2)
             throw std::runtime_error(std::string(argv[0]) + " <input_file> [<output_dir>]");
 
         const std::string wac_path = argv[1];
-        const std::string output_path = (argc == 3) ? argv[2] : wac_path + "/extracted";
+        int last_slash = -1;
```

Comment:
not entirely sure what this code is doing but can almost definitely be done with https://en.cppreference.com/w/cpp/filesystem/path

put
```cpp
using fs = std::filesystem
```
on the top of the file
and make the paths in here `fs::path` instead of string. then use appropriate methods on that

Reply:
I really dislike the using keyword, especially for small projects like this, so i'll just write out std::filesystem instead. makes it way more clear